### PR TITLE
FM2-443: AuthenticationFilter to allow requests with other authentication schemes to go through

### PR DESCRIPTION
**Pull request by Ruhanga**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-443

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java [hunk 1]
```diff
@@ -58,10 +58,8 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 							return;
 						}
 					} else {
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java:60]: @Ruhanga what happens for this else after removing the bellow lines

Comment (Ruhanga) [reply omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java:60]: @HerbertYiga, the request is allowed through to other filters.

**Review by Ruhanga — COMMENTED** [review]
Hi @ibacher, a kind reminder to review and if all is okay, to merge this PR. Thanks.

**Review by ibacher — APPROVED** [review]

File: omod/src/test/java/org/openmrs/module/fhir2/web/authentication/RequireAuthenticationInterceptorTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		
+	}
+	
+	@Test
+	public void ensureUserAuthenticated_shouldReturnTrueGivenWellKnownUri() throws Exception {
+		// setup
+		request.setRequestURI("/.well-known");
+		
+		// replay and verify
+		assertTrue(interceptor.ensureUserAuthenticated(request, response));
```
Comment (ibacher) [inline omod/src/test/java/org/openmrs/module/fhir2/web/authentication/RequireAuthenticationInterceptorTest.java:57]: Actually @Ruhanga, once very small change here: could we re-write these to use Hamcrest to be consistent with the other tests in the module, e.g.

```suggestion
		assertThat(interceptor.ensureUserAuthenticated(request, response), is(true));
```

Comment (Ruhanga) [reply omod/src/test/java/org/openmrs/module/fhir2/web/authentication/RequireAuthenticationInterceptorTest.java]: Thanks @ibacher, done.

**Review by mozzy11 — APPROVED** [review]

**Pull request merged**